### PR TITLE
Allow LDAP search users with *term*

### LINKED ISSUE #11540: User search not efficient, returns partial matches
### Steps to reproduce
1. Go to the user section
2. Search for part of the full name or username

### Expected behaviour
I should find users with parts of there full name or there username

### Actual behaviour
I only find user when i start with the full name, if i just enter the lastname or the username I don't find anything

### Server configuration

**Operating system**:CentOS

**Web server:** NginX

**Database:** MariaDB

**PHP version:** 7.1

**Nextcloud version:** 13.0.4

**Updated from an older Nextcloud/ownCloud or fresh install:** updated

**Where did you install Nextcloud from:** nextcloud.com

### PR DESCRIPTION
**Pull request by smarinier**
Fix #11540 and #14153

* Resolves: # 

Fix #11540 and #14153
This issues were close, but because the user "didn't use NextCloud anymore"

## Summary

In Database User Backend, search is based on "*term*". But in LDAP, it's only "term*". This is confusing for user experience.

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ X] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [0] Screenshots before/after for front-end changes
       No front change
- [0] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
   This behaviour is not described in documentation

- [0] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Comment by smarinier** [comment]
My commit is not in the context of Arawa company. How can i have two different sign-offs ?

File: apps/user_ldap/lib/Access.php [hunk 1]
```diff
@@ -1564,7 +1564,7 @@ private function prepareSearchTerm(string $term): string {
 		if ($term === '') {
 			$result = '*';
 		} elseif ($allowEnum !== 'no') {
-			$result = $term . '*';
+			$result = '*' . $term . '*';
```
Comment (artonge) [inline apps/user_ldap/lib/Access.php:1567]: I think the limitation was on purpose. @come-nc and @blizzz?

Comment (blizzz) [reply apps/user_ldap/lib/Access.php:1567]: It is on purpose, yes.

**Comment by artonge** [comment]
> My commit is not in the context of Arawa company. How can i have two different sign-offs ?

Regarding DCO, I think it has to do with your local git config

File: apps/user_ldap/lib/Access.php (hunk 1 above)
Comment (smarinier) [reply apps/user_ldap/lib/Access.php:1567]: I read some supposed performance limitations on big LDAP organisations. But:
 - all i read on LDAP documentation never mention "don't do *term*" as it is even described in the main documentation.
 - this might be based on old benchmarks on bad LDAP system
 - why is the "*" allowed that will send "all users" in that case ?
 - the SQL part does the *term* and nobody asked to remove this for performance reason
 - this is very very confusing for the user experience (i thought this was a bug and i lost hours searching what was wrong with my LDAP until i debug in PHP) 
 
 In conclusion, at least, why presuming all ldap are made with millions of users ? Why not let users have the same behaviour between LDAP and NextCloud ? And maybe bring an LDAP option "don't allow *term* because my ldap is too slow" if someone complain for it ?

**Comment by smarinier** [comment]
> > My commit is not in the context of Arawa company. How can i have two different sign-offs ?
> 
> Regarding DCO, I think it has to do with your local git config

Are you sure ? my git config does not have any information like my name. I'm registered only with my github account. On the other hand, this github account is configured with my two emails. I guess that there is something in your DCO telling smarinier is sebastien.marinier@arawa.fr and it refuses another email (as i was refused once because i use my real accentuated name "Sébastien" instead of "Sebastien", with the correct email).

**Comment by artonge** [comment]
> Are you sure ?

No. Can you run the following from the nextcloud repo:

```shell
git config user.email
git config --global user.email
```

**Comment by smarinier** [comment]
> > Are you sure ?
> 
> No. Can you run the following from the nextcloud repo:
> 
> ```shell
> git config user.email
> git config --global user.email
> ```

Dawn! It was in the global. I've changed it and re-push the sign off. Thanks

**Review by come-nc — COMMENTED** [review]
If you make it a config option, maybe, but we cannot merge this as-is.
I do not have the details about when the decision was first taken, and I agree that with a good LDAP setup wildcard search is not an issue, but there is a lot of code in Nextcloud which is made to account for badly configured LDAP server, this is not the only place.
